File: gtmworks_ui/src/services/SolutionService.js
import http from "../http-common"; 

class SolutionService {
  getAllSolutions(searchDTO) {
    console.log(searchDTO)
    return this.getRequest(`/solution/solutions`, searchDTO);
  }

  get(solutionId) {
    return this.getRequest(`/solution/${solutionId}`, null);
  }

  findByField(matchData) {
    return this.getRequest(`/solution?field=${matchData}`, null);
  }

  addSolution(data) {
    return http.post("/solution/addSolution", data);
  }

  update(data) {
  	return http.post("/solution/updateSolution", data);
  }
  
  uploadImage(data,solutionId) {
  	return http.postForm("/solution/uploadImage/"+solutionId, data);
  }




	postRequest(url, data) {
		return http.post(url, data);
      };

	getRequest(url, params) {
        return http.get(url, {
        	params: params
        });
    };

}

export default new SolutionService();

File: src/main/java/com/gtmworks/service/SolutionService.java
package com.gtmworks.service;

import java.util.List;

import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.data.jpa.domain.Specification;
import org.springframework.http.ResponseEntity;

import com.gtmworks.domain.Solution;
import com.gtmworks.dto.SolutionDTO;
import com.gtmworks.dto.SolutionSearchDTO;
import com.gtmworks.dto.SolutionPageDTO;
import com.gtmworks.dto.SolutionConvertCriteriaDTO;
import com.gtmworks.service.GenericService;
import com.gtmworks.dto.common.RequestDTO;
import com.gtmworks.dto.common.ResultDTO;
import java.util.List;
import java.util.Optional;





public interface SolutionService extends GenericService<Solution, Integer> {

	List<Solution> findAll();

	ResultDTO addSolution(SolutionDTO solutionDTO, RequestDTO requestDTO);

	ResultDTO updateSolution(SolutionDTO solutionDTO, RequestDTO requestDTO);

    Page<Solution> getAllSolutions(Pageable pageable);

    Page<Solution> getAllSolutions(Specification<Solution> spec, Pageable pageable);

	ResponseEntity<SolutionPageDTO> getSolutions(SolutionSearchDTO solutionSearchDTO);
	
	List<SolutionDTO> convertSolutionsToSolutionDTOs(List<Solution> solutions, SolutionConvertCriteriaDTO convertCriteria);

	SolutionDTO getSolutionDTOById(Integer solutionId);







}





